Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -19,6 +19,73 @@ This file sets up the basic `positivity` extensions tagged with the `@[positivit
 namespace Mathlib.Meta.Positivity
 open Lean Meta Qq Function
 
+section ite
+variable [Zero α] (p : Prop) [Decidable p] {a b : α}
+
+private lemma ite_pos [LT α] (ha : 0 < a) (hb : 0 < b) : 0 < ite p a b :=
+by by_cases p <;> simp [*]
+
+private lemma ite_nonneg [LE α] (ha : 0 ≤ a) (hb : 0 ≤ b) : 0 ≤ ite p a b :=
+by by_cases p <;> simp [*]
+
+private lemma ite_nonneg_of_pos_of_nonneg [Preorder α] (ha : 0 < a) (hb : 0 ≤ b) : 0 ≤ ite p a b :=
+ite_nonneg _ ha.le hb
+
+private lemma ite_nonneg_of_nonneg_of_pos [Preorder α] (ha : 0 ≤ a) (hb : 0 < b) : 0 ≤ ite p a b :=
+ite_nonneg _ ha hb.le
+
+private lemma ite_ne_zero (ha : a ≠ 0) (hb : b ≠ 0) : ite p a b ≠ 0 := by by_cases p <;> simp [*]
+
+private lemma ite_ne_zero_of_pos_of_ne_zero [Preorder α] (ha : 0 < a) (hb : b ≠ 0) :
+  ite p a b ≠ 0 :=
+ite_ne_zero _ ha.ne' hb
+
+private lemma ite_ne_zero_of_ne_zero_of_pos [Preorder α] (ha : a ≠ 0) (hb : 0 < b) :
+  ite p a b ≠ 0 :=
+ite_ne_zero _ ha hb.ne'
+
+end ite
+
+/-- The `positivity` extension which identifies expressions of the form `ite p a b`,
+such that `positivity` successfully recognises both `a` and `b`. -/
+@[positivity ite _ _] def evalIte : PositivityExt where eval {u α} zα pα e := do
+  let .app (.app (.app f (p : Q(Prop))) (a : Q($α))) (b : Q($α)) ← withReducible (whnf e)
+    | throwError "not ite"
```

Comment:
I think I messed up this pattern-matching, but I can't figure out how to fix it. @digama0, would you enlighten me?

Reply:
This needed an additional `.app`, to pick up the `Decidable` instance.